cnd/src/network/start_swap.rs

```diff
@@ -30,8 +30,7 @@ where
             RemoteData {
                 ethereum_identity: Some(ethereum_identity),
                 lightning_identity: Some(lightning_identity),
-                // Do not make this None, secret_hash is in the behaviour event for Alice.
```

Comment:
Can we have a ticket for this comment please, this needs fixing in the network behaviour.

Reply:
The bigger problem here is that we model `RemoteData` vs `LocalData` whereas - I think - what we should actually be doing is modelling this similar to the DB: `WhatAliceSharesWithBob` and `WhatBobSharesWithAlice`.

Interestingly, `WhatAliceLearnedFromBob` == `WhatBobSharesWithAlice` :smile: 
So there is some interesting modelling work to be done here :)

I attempted this refactoring already once but the `Swaps` struct is too entangled with the current model to do this quickly unfortunately.